helper/call.go

```diff
@@ -862,3 +827,126 @@ func (c *ContractCaller) GetCheckpointSign(txHash common.Hash) ([]byte, []byte,
 
 	return UnpackSigAndVotes(payload, abi)
 }
+
+// utility and helper methods
+
+// NewLru create instance of lru
+func NewLru(size int) (*lru.Cache, error) {
+	lruObj, err := lru.New(size)
+	if err != nil {
+		return nil, err
+	}
+
+	return lruObj, nil
+}
+
+// populateABIs fills the package level cache for contracts' ABIs
+// When called the first time, ContractsABIsMap will be filled and getABI method won't be invoked the next times
+// This reduces the number of calls to json decode methods made by the contract caller
+// It uses ABIs' names instead of contracts addresses, as the latter might not be available at init time
+func populateABIs(contractCallerObj *ContractCaller) error {
+	var err error
+
+	if ContractsABIsMap[RootChainABI] == nil {
```

Comment:
Could we have an array of [RootChainABI, StakingInfoABI, ...] and just iterate over them and do the caching and checking things? I believe it might remove a lot of code duplication.

Reply:
Done, please resolve if appropriate @JekaMas